File: MyAppL11Assign/app/src/main/java/com/example/dell/myappl11assign/adapters/CommentAdapter.java
package com.example.dell.myappl11assign.adapters;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.dell.myappl11assign.R;
import com.example.dell.myappl11assign.models.Comment;

import java.util.ArrayList;

/**
 * Created by dell on 01-07-2017.
 */

public class CommentAdapter extends RecyclerView.Adapter<CommentAdapter.CommentViewHolder> {
    public static final String TAG = "CommentAdapter";
    private Context context;
    private ArrayList<Comment> comments;

    public CommentAdapter(Context context, ArrayList<Comment> comments) {
        this.context = context;
        this.comments = comments;
    }

    public void updateComments(ArrayList<Comment> comments) {
        this.comments = comments;
        notifyDataSetChanged();
    }

    @Override
    public CommentViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater li = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View itemView = li.inflate(R.layout.list_item_comment, parent, false);
        return new CommentViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(CommentViewHolder holder, int position) {
        Comment thisComment = comments.get(position);
        holder.tvCommentBody.setText(thisComment.getBody());
        holder.tvCommentEmail.setText(thisComment.getEmail());
        holder.tvCommentName.setText(thisComment.getName());
    }

    @Override
    public int getItemCount() {
        return comments.size();
    }

    class CommentViewHolder extends RecyclerView.ViewHolder {
        TextView tvCommentBody, tvCommentName, tvCommentEmail;

        public CommentViewHolder(View itemView) {
            super(itemView);
            tvCommentBody = (TextView) itemView.findViewById(R.id.tvCommentBody);
            tvCommentEmail = (TextView) itemView.findViewById(R.id.tvCommentEmail);
            tvCommentName = (TextView) itemView.findViewById(R.id.tvCommentName);
        }
    }
}

File: AndroidSample_Retrofit_RESTAPI/app/src/main/java/com/codingblocks/restapiretrofitjson/adapters/CommentAdapter.kt
package com.codingblocks.restapiretrofitjson.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.codingblocks.restapiretrofitjson.R
import com.codingblocks.restapiretrofitjson.models.Comment
import kotlinx.android.synthetic.main.list_item_comment.view.*
import java.util.*

/**
 * Created by championswimmer on 29/06/17.
 */

class CommentAdapter(private val context: Context,
                     private var comments: ArrayList<Comment>)
    : RecyclerView.Adapter<CommentAdapter.CommentViewHolder>() {

    fun updateComments(comments: ArrayList<Comment>) {
        this.comments = comments;
        notifyDataSetChanged();
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CommentViewHolder {


        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.list_item_comment, parent, false)

        return CommentViewHolder(
                itemView.tvCommentBody,
                itemView.tvCommentName,
                itemView.tvCommentEmail,
                itemView
        )
    }

    override fun onBindViewHolder(holder: CommentViewHolder, position: Int) {

        val (_, _, name, email, body) = comments[position]
        holder.tvCommentBody.text = body
        holder.tvCommentName.text = name
        holder.tvCommentEmail.text = email

    }

    override fun getItemCount(): Int {
        return comments.size
    }

    class CommentViewHolder(
            public val tvCommentBody: TextView,
            public val tvCommentName: TextView,
            public val tvCommentEmail: TextView,
            itemView: View
    ) : RecyclerView.ViewHolder(itemView) {


    }
}
